src/lib.rs

```diff
@@ -55,14 +54,12 @@ extern crate env_logger;
 
 #[cfg(test)]
 extern crate rand;
-#[cfg(test)]
-extern crate test;
 
 mod file_set;
-mod index;
+pub mod index;
```

Comment:
Is exposing the index and segment modules just for the benchmarks? I'm not sure I want to expose these to the public API quite yet.

Reply:
You are right, its public just for the benchmarks. It is not possible to benchmark non-pub functions. External benchmarks, including those using Criterion.rs, are compiled as a separate crate, and non-pub functions are not visible to the benchmarks.
I will try to figure a way around the limitation and will update my PR afterwards.

Reply:
I have added `bench-private` flag wich will enable benchmarks for the private api. I'm not totally happy with it but was unable to find a better solution.

Reply:
That's fair. Baring a better alternative, this should be fine. I'll consider exposing these publicly with a better API in the next version.